firebase-installations/src/main/java/com/google/firebase/installations/remote/ResponseHandler.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.remote;
+
+import androidx.annotation.GuardedBy;
+import com.google.firebase.installations.Utils;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The {@link ResponseHandler} class calculates the next allowed request time. Also, decides whether
+ * a network request to FIS servers is allowed to execute.
+ *
+ * @hide
+ */
+class ResponseHandler {
+  private static final long BACKOFF_TIME_24H_IN_MILLIS = TimeUnit.HOURS.toMillis(24);
+  private static final long BACKOFF_TIME_30_MINS_IN_MILLIS = TimeUnit.MINUTES.toMillis(30);
+  private final Utils utils;
+
+  @GuardedBy("this")
+  private long nextAllowedRequestTime = Long.MAX_VALUE;
+
+  @GuardedBy("this")
+  private int attemptCount = 0;
+
+  ResponseHandler(Utils utils) {
+    this.utils = utils;
+  }
+
+  public synchronized void setNextAllowedRequestTime(int responseCode) {
+    if (responseCode == 200) {
+      resetAllowedRequestTime();
+      return;
+    }
+    attemptCount++;
+    long backOffTime = getBackoffTime(responseCode);
+    nextAllowedRequestTime = utils.currentTimeInMillis() + backOffTime;
+  }
+
+  private synchronized void resetAllowedRequestTime() {
+    nextAllowedRequestTime = Long.MAX_VALUE;
```

Comment:
Why do we set this at all?
Can we just ignore this value or set it to `-1`?

It seems like you are abusing some existing constant (i.e. `Long.MAX_VALUE`) for semantic purposes.
If that is the case, I recommend to rather introduce your own constant that has a speaking name.
Example: `NEXT_REQUEST_TIME_IMMEDIATELY` or something similar.

Reply:
Agreed on ignoring the value instead of resetting it. I initially had a thought without resetting it will not allow some requests to FIS servers. But on rethinking the logic, it will work correctly without resetting the nextRequestTime.
Thanks for pointing it out. PTAL.